scene/gui/file_dialog.cpp

```diff
@@ -434,14 +435,17 @@ void FileDialog::update_file_list() {
 
 			if (file->get_text()==files.front()->get())
 				ti->select(0);
+			else if (match_str==files.front()->get())
+				ti->select(0);
 		}
```

Comment:
maybe

`if (file->get_text()==files.front()->get() || match_str==files.front()->get())`

as now it suggest that ifs execute different block of code